src/main/java/org/embulk/util/config/Compat.java

```diff
@@ -178,7 +178,27 @@ private static Method getToJsonMethod() {
             // A method instance retrieved from the public interface "org.embulk.config.DataSource" would solve the problem.
             return DataSource.class.getMethod("toJson");
         } catch (final NoSuchMethodException ex) {
-            // Expected: toJson is not defined in Embulk v0.10.2 or earlier.
+            // Expected: toJson is not defined in "org.embulk.config.DataSource" when a user is running
+            // Embulk v0.10.2 or earlier.
+            //
+            // Even in the case, the received DataSource instance can still be of embulk-util-config's
+            // "org.embulk.util.config.DataSourceImpl" from another plugin (ex. input <=> parser), or
+            // from itself. As "org.embulk.util.config.DataSourceImpl" does not have "getObjectNode",
+            // it must still be rebuilt with "toJson" retrieved in some way.
+            //
+            // Pass-through to the next trial to retrieve the "toJson" method, then.
+        }
+
+        final Class<? extends DataSource> dataSourceImplClass = source.getClass();
+        try {
+            // Getting the "toJson" method from the implementation class embulk-core's "org.embulk.config.DataSourceImpl",
+            // or embulk-util-config's "org.embulk.util.config.DataSourceImpl".
+            return dataSourceImplClass.getMethod("toJson");
```

Comment:
Just wanted to clarify if I understand correctly: there is still a chance that this will throw an IllegalAccessException which is treated in https://github.com/embulk/embulk-util-config/pull/26?

Reply:
Yes. If another plugin is using `embulk-util-config:0.1.4` or earlier, and a plugin using this `embulk-util-config:0.2.1` receives a processed `ConfigSource/TaskSource` from the other plugin, it can happen.
(Once everyone starts using `embulk-util-config:0.2.1` and/or Embulk v0.10.3+, it should never happen.)